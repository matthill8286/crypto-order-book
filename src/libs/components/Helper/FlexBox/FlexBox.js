function _extends() { _extends = Object.assign || function (target) { for (var i = 1; i < arguments.length; i++) { var source = arguments[i]; for (var key in source) { if (Object.prototype.hasOwnProperty.call(source, key)) { target[key] = source[key]; } } } return target; }; return _extends.apply(this, arguments); }

function _objectWithoutProperties(source, excluded) { if (source == null) return {}; var target = _objectWithoutPropertiesLoose(source, excluded); var key, i; if (Object.getOwnPropertySymbols) { var sourceSymbolKeys = Object.getOwnPropertySymbols(source); for (i = 0; i < sourceSymbolKeys.length; i++) { key = sourceSymbolKeys[i]; if (excluded.indexOf(key) >= 0) continue; if (!Object.prototype.propertyIsEnumerable.call(source, key)) continue; target[key] = source[key]; } } return target; }

function _objectWithoutPropertiesLoose(source, excluded) { if (source == null) return {}; var target = {}; var sourceKeys = Object.keys(source); var key, i; for (i = 0; i < sourceKeys.length; i++) { key = sourceKeys[i]; if (excluded.indexOf(key) >= 0) continue; target[key] = source[key]; } return target; }

import React from 'react';
import { css, styled } from "../../../styles/styled";
var StyledBox = styled.div.withConfig({
  displayName: "FlexBox__StyledBox",
  componentId: "sc-52meld-0"
})(["", " display:", ";flex-direction:", ";", ";", ";", ";", ";"], function (_ref) {
  var debug = _ref.debug;
  return debug && css(["border:1px solid green;padding:8px;"]);
}, function (_ref2) {
  var display = _ref2.display;
  return display;
}, function (_ref3) {
  var flexDirection = _ref3.flexDirection;
  return flexDirection;
}, function (_ref4) {
  var flexWrap = _ref4.flexWrap;
  return flexWrap && "flex-wrap: ".concat(flexWrap);
}, function (_ref5) {
  var justifyContent = _ref5.justifyContent;
  return justifyContent && "justify-content: ".concat(justifyContent);
}, function (_ref6) {
  var alignContent = _ref6.alignContent;
  return alignContent && "align-content: ".concat(alignContent);
}, function (_ref7) {
  var alignItems = _ref7.alignItems;
  return alignItems && "align-items: ".concat(alignItems);
});
export var FlexBox = function FlexBox(props) {
  var _props$debug = props.debug,
      debug = _props$debug === void 0 ? false : _props$debug,
      _props$display = props.display,
      display = _props$display === void 0 ? 'flex' : _props$display,
      _props$flexDirection = props.flexDirection,
      flexDirection = _props$flexDirection === void 0 ? 'row' : _props$flexDirection,
      flexWrap = props.flexWrap,
      justifyContent = props.justifyContent,
      alignContent = props.alignContent,
      alignItems = props.alignItems,
      children = props.children,
      otherProps = _objectWithoutProperties(props, ["debug", "display", "flexDirection", "flexWrap", "justifyContent", "alignContent", "alignItems", "children"]);

  return React.createElement(StyledBox, _extends({
    debug: debug,
    display: display,
    flexDirection: flexDirection,
    flexWrap: flexWrap,
    justifyContent: justifyContent,
    alignContent: alignContent,
    alignItems: alignItems
  }, otherProps), children);
};
FlexBox.displayName = 'FlexBox';
var StyledFlexItem = styled.div.withConfig({
  displayName: "FlexBox__StyledFlexItem",
  componentId: "sc-52meld-1"
})(["", " order:", ";flex:", ";align-self:", ";padding:", ";"], function (_ref8) {
  var debug = _ref8.debug;
  return debug && css(["border:1px solid blue;padding:8px;"]);
}, function (_ref9) {
  var order = _ref9.order;
  return order;
}, function (_ref10) {
  var flex = _ref10.flex;
  return flex;
}, function (_ref11) {
  var alignSelf = _ref11.alignSelf;
  return alignSelf;
}, function (_ref12) {
  var theme = _ref12.theme,
      padding = _ref12.padding;
  return padding ? "0 ".concat(theme.spacing.base.xs) : 0;
});
export var FlexItem = function FlexItem(props) {
  var children = props.children,
      debug = props.debug,
      _props$order = props.order,
      order = _props$order === void 0 ? 0 : _props$order,
      _props$flex = props.flex,
      flex = _props$flex === void 0 ? '0 1 auto' : _props$flex,
      _props$alignSelf = props.alignSelf,
      alignSelf = _props$alignSelf === void 0 ? 'auto' : _props$alignSelf,
      otherProps = _objectWithoutProperties(props, ["children", "debug", "order", "flex", "alignSelf"]);

  return React.createElement(StyledFlexItem, _extends({
    debug: debug,
    order: order,
    flex: flex,
    alignSelf: alignSelf
  }, otherProps), children);
};
FlexItem.displayName = 'FlexItem';
//# sourceMappingURL=FlexBox.js.map