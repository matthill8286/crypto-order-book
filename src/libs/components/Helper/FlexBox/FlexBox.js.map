{"version":3,"sources":["../../../../src/components/Helper/FlexBox/FlexBox.tsx"],"names":["React","css","styled","StyledBox","div","debug","display","flexDirection","flexWrap","justifyContent","alignContent","alignItems","FlexBox","props","children","otherProps","displayName","StyledFlexItem","order","flex","alignSelf","theme","padding","spacing","base","xs","FlexItem"],"mappings":";;;;;;AAAA,OAAOA,KAAP,MAA0B,OAA1B;AAEA,SAASC,GAAT,EAAcC,MAAd;AAoBA,IAAMC,SAAS,GAAGD,MAAM,CAACE,GAAV;AAAA;AAAA;AAAA,mEACX;AAAA,MAAGC,KAAH,QAAGA,KAAH;AAAA,SACAA,KAAK,IACLJ,GADK,yCADL;AAAA,CADW,EAOF;AAAA,MAAGK,OAAH,SAAGA,OAAH;AAAA,SAAiBA,OAAjB;AAAA,CAPE,EAQK;AAAA,MAAGC,aAAH,SAAGA,aAAH;AAAA,SAAuBA,aAAvB;AAAA,CARL,EASX;AAAA,MAAGC,QAAH,SAAGA,QAAH;AAAA,SAAkBA,QAAQ,yBAAkBA,QAAlB,CAA1B;AAAA,CATW,EAUX;AAAA,MAAGC,cAAH,SAAGA,cAAH;AAAA,SAAwBA,cAAc,+BAAwBA,cAAxB,CAAtC;AAAA,CAVW,EAWX;AAAA,MAAGC,YAAH,SAAGA,YAAH;AAAA,SAAsBA,YAAY,6BAAsBA,YAAtB,CAAlC;AAAA,CAXW,EAYX;AAAA,MAAGC,UAAH,SAAGA,UAAH;AAAA,SAAoBA,UAAU,2BAAoBA,UAApB,CAA9B;AAAA,CAZW,CAAf;AAeA,OAAO,IAAMC,OAAyB,GAAG,SAA5BA,OAA4B,CAAAC,KAAK,EAAI;AAAA,qBAW5CA,KAX4C,CAE9CR,KAF8C;AAAA,MAE9CA,KAF8C,6BAEtC,KAFsC;AAAA,uBAW5CQ,KAX4C,CAG9CP,OAH8C;AAAA,MAG9CA,OAH8C,+BAGpC,MAHoC;AAAA,6BAW5CO,KAX4C,CAI9CN,aAJ8C;AAAA,MAI9CA,aAJ8C,qCAI9B,KAJ8B;AAAA,MAK9CC,QAL8C,GAW5CK,KAX4C,CAK9CL,QAL8C;AAAA,MAM9CC,cAN8C,GAW5CI,KAX4C,CAM9CJ,cAN8C;AAAA,MAO9CC,YAP8C,GAW5CG,KAX4C,CAO9CH,YAP8C;AAAA,MAQ9CC,UAR8C,GAW5CE,KAX4C,CAQ9CF,UAR8C;AAAA,MAS9CG,QAT8C,GAW5CD,KAX4C,CAS9CC,QAT8C;AAAA,MAU3CC,UAV2C,4BAW5CF,KAX4C;;AAahD,SACE,oBAAC,SAAD;AACE,IAAA,KAAK,EAAER,KADT;AAEE,IAAA,OAAO,EAAEC,OAFX;AAGE,IAAA,aAAa,EAAEC,aAHjB;AAIE,IAAA,QAAQ,EAAEC,QAJZ;AAKE,IAAA,cAAc,EAAEC,cALlB;AAME,IAAA,YAAY,EAAEC,YANhB;AAOE,IAAA,UAAU,EAAEC;AAPd,KAQMI,UARN,GASGD,QATH,CADF;AAaD,CA1BM;AA4BPF,OAAO,CAACI,WAAR,GAAsB,SAAtB;AAUA,IAAMC,cAAc,GAAGf,MAAM,CAACE,GAAV;AAAA;AAAA;AAAA,gEAChB;AAAA,MAAGC,KAAH,SAAGA,KAAH;AAAA,SACAA,KAAK,IACLJ,GADK,wCADL;AAAA,CADgB,EAOT;AAAA,MAAGiB,KAAH,SAAGA,KAAH;AAAA,SAAeA,KAAf;AAAA,CAPS,EAQV;AAAA,MAAGC,IAAH,UAAGA,IAAH;AAAA,SAAcA,IAAd;AAAA,CARU,EASJ;AAAA,MAAGC,SAAH,UAAGA,SAAH;AAAA,SAAmBA,SAAnB;AAAA,CATI,EAUP;AAAA,MAAGC,KAAH,UAAGA,KAAH;AAAA,MAAUC,OAAV,UAAUA,OAAV;AAAA,SAAyBA,OAAO,eAAQD,KAAK,CAACE,OAAN,CAAcC,IAAd,CAAmBC,EAA3B,IAAkC,CAAlE;AAAA,CAVO,CAApB;AAaA,OAAO,IAAMC,QAA2B,GAAG,SAA9BA,QAA8B,CAAAb,KAAK,EAAI;AAAA,MAC1CC,QAD0C,GAC2CD,KAD3C,CAC1CC,QAD0C;AAAA,MAChCT,KADgC,GAC2CQ,KAD3C,CAChCR,KADgC;AAAA,qBAC2CQ,KAD3C,CACzBK,KADyB;AAAA,MACzBA,KADyB,6BACjB,CADiB;AAAA,oBAC2CL,KAD3C,CACdM,IADc;AAAA,MACdA,IADc,4BACP,UADO;AAAA,yBAC2CN,KAD3C,CACKO,SADL;AAAA,MACKA,SADL,iCACiB,MADjB;AAAA,MAC4BL,UAD5B,4BAC2CF,KAD3C;;AAGlD,SACE,oBAAC,cAAD;AAAgB,IAAA,KAAK,EAAER,KAAvB;AAA8B,IAAA,KAAK,EAAEa,KAArC;AAA4C,IAAA,IAAI,EAAEC,IAAlD;AAAwD,IAAA,SAAS,EAAEC;AAAnE,KAAkFL,UAAlF,GACGD,QADH,CADF;AAKD,CARM;AAUPY,QAAQ,CAACV,WAAT,GAAuB,UAAvB","sourcesContent":["import React, { FC } from 'react'\nimport { media } from '@/styles/media'\nimport { css, styled } from '@/styles/styled'\n\nexport type FlexAlignItemsOptions =\n  | 'flex-start'\n  | 'flex-end'\n  | 'center'\n  | 'space-between'\n  | 'space-around'\n  | 'stretch'\n\nexport interface FlexBoxProps {\n  debug?: boolean\n  display?: 'flex' | 'inline-flex'\n  flexDirection?: 'row' | 'row-reverse' | 'column' | 'column-reverse'\n  flexWrap?: 'nowrap' | 'wrap' | 'wrap-reverse'\n  justifyContent?: 'flex-start' | 'flex-end' | 'center' | 'space-between' | 'space-around'\n  alignContent?: 'flex-start' | 'flex-end' | 'center' | 'baseline' | 'stretch'\n  alignItems?: FlexAlignItemsOptions\n}\n\nconst StyledBox = styled.div<FlexBoxProps>`\n  ${({ debug }) =>\n    debug &&\n    css`\n      border: 1px solid green;\n      padding: 8px;\n    `}\n  display: ${({ display }) => display};\n  flex-direction: ${({ flexDirection }) => flexDirection};\n  ${({ flexWrap }) => flexWrap && `flex-wrap: ${flexWrap}`};\n  ${({ justifyContent }) => justifyContent && `justify-content: ${justifyContent}`};\n  ${({ alignContent }) => alignContent && `align-content: ${alignContent}`};\n  ${({ alignItems }) => alignItems && `align-items: ${alignItems}`};\n`\n\nexport const FlexBox: FC<FlexBoxProps> = props => {\n  const {\n    debug = false,\n    display = 'flex',\n    flexDirection = 'row',\n    flexWrap,\n    justifyContent,\n    alignContent,\n    alignItems,\n    children,\n    ...otherProps\n  } = props\n\n  return (\n    <StyledBox\n      debug={debug}\n      display={display}\n      flexDirection={flexDirection}\n      flexWrap={flexWrap}\n      justifyContent={justifyContent}\n      alignContent={alignContent}\n      alignItems={alignItems}\n      {...otherProps}>\n      {children}\n    </StyledBox>\n  )\n}\n\nFlexBox.displayName = 'FlexBox'\n\nexport interface FlexItemProps {\n  debug?: boolean\n  order?: number\n  flex?: string\n  alignSelf?: 'auto' | 'flex-start' | 'flex-end' | 'center' | 'baseline' | 'stretch'\n  padding?: boolean\n}\n\nconst StyledFlexItem = styled.div<FlexItemProps>`\n  ${({ debug }) =>\n    debug &&\n    css`\n      border: 1px solid blue;\n      padding: 8px;\n    `}\n  order: ${({ order }) => order};\n  flex: ${({ flex }) => flex};\n  align-self: ${({ alignSelf }) => alignSelf};\n  padding: ${({ theme, padding }) => (padding ? `0 ${theme.spacing.base.xs}` : 0)};\n`\n\nexport const FlexItem: FC<FlexItemProps> = props => {\n  const { children, debug, order = 0, flex = '0 1 auto', alignSelf = 'auto', ...otherProps } = props\n\n  return (\n    <StyledFlexItem debug={debug} order={order} flex={flex} alignSelf={alignSelf} {...otherProps}>\n      {children}\n    </StyledFlexItem>\n  )\n}\n\nFlexItem.displayName = 'FlexItem'\n"],"file":"FlexBox.js"}