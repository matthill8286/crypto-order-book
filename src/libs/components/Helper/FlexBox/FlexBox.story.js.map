{"version":3,"sources":["../../../../src/components/Helper/FlexBox/FlexBox.story.tsx"],"names":["boolean","number","select","text","storiesOf","React","FlexBox","FlexItem","readme","module","add","flexBoxKnobs","debug","display","flexDirection","flexWrap","justifyContent","alignContent","alignItems","firstFlexItemProps","order","flex","alignSelf","secondFlexItemProps","info"],"mappings":"AAAA,SAASA,OAAT,EAAkBC,MAAlB,EAA0BC,MAA1B,EAAkCC,IAAlC,QAA8C,wBAA9C;AACA,SAASC,SAAT,QAA0B,kBAA1B;AACA,OAAOC,KAAP,MAAkB,OAAlB;AACA,SAASC,OAAT,EAAgCC,QAAhC;AACA,OAAOC,MAAP;AAEAJ,SAAS,CAAC,8BAAD,EAAiCK,MAAjC,CAAT,CAAkDC,GAAlD,CACE,SADF,EAEE,YAAM;AACJ,MAAMC,YAA0B,GAAG;AACjCC,IAAAA,KAAK,EAAEZ,OAAO,CAAC,WAAD,EAAc,IAAd,CADmB;AAEjCa,IAAAA,OAAO,EAAEX,MAAM,CAAC,aAAD,EAAgB,CAAC,MAAD,EAAS,aAAT,CAAhB,EAAyC,MAAzC,CAFkB;AAGjCY,IAAAA,aAAa,EAAEZ,MAAM,CACnB,mBADmB,EAEnB,CAAC,KAAD,EAAQ,aAAR,EAAuB,QAAvB,EAAiC,gBAAjC,CAFmB,EAGnB,KAHmB,CAHY;AAQjCa,IAAAA,QAAQ,EAAEb,MAAM,CAAC,cAAD,EAAiB,CAAC,QAAD,EAAW,MAAX,EAAmB,cAAnB,CAAjB,EAAqD,MAArD,CARiB;AASjCc,IAAAA,cAAc,EAAEd,MAAM,CACpB,oBADoB,EAEpB,CAAC,YAAD,EAAe,UAAf,EAA2B,QAA3B,EAAqC,eAArC,EAAsD,cAAtD,CAFoB,EAGpB,YAHoB,CATW;AAcjCe,IAAAA,YAAY,EAAEf,MAAM,CAClB,kBADkB,EAElB,CAAC,YAAD,EAAe,UAAf,EAA2B,QAA3B,EAAqC,UAArC,EAAiD,SAAjD,CAFkB,EAGlB,YAHkB,CAda;AAmBjCgB,IAAAA,UAAU,EAAEhB,MAAM,CAChB,gBADgB,EAEhB,CAAC,YAAD,EAAe,UAAf,EAA2B,QAA3B,EAAqC,eAArC,EAAsD,cAAtD,EAAsE,SAAtE,CAFgB,EAGhB,YAHgB;AAnBe,GAAnC;AAyBA,MAAMiB,kBAAiC,GAAG;AACxCP,IAAAA,KAAK,EAAEZ,OAAO,CAAC,kBAAD,EAAqB,IAArB,CAD0B;AAExCoB,IAAAA,KAAK,EAAEnB,MAAM,CAAC,kBAAD,EAAqB,CAArB,CAF2B;AAGxCoB,IAAAA,IAAI,EAAElB,IAAI,CAAC,iBAAD,EAAoB,UAApB,CAH8B;AAIxCmB,IAAAA,SAAS,EAAEpB,MAAM,CACf,sBADe,EAEf,CAAC,MAAD,EAAS,YAAT,EAAuB,UAAvB,EAAmC,QAAnC,EAA6C,UAA7C,EAAyD,SAAzD,CAFe,EAGf,MAHe;AAJuB,GAA1C;AAUA,MAAMqB,mBAAkC,GAAG;AACzCX,IAAAA,KAAK,EAAEZ,OAAO,CAAC,mBAAD,EAAsB,IAAtB,CAD2B;AAEzCoB,IAAAA,KAAK,EAAEnB,MAAM,CAAC,mBAAD,EAAsB,CAAtB,CAF4B;AAGzCoB,IAAAA,IAAI,EAAElB,IAAI,CAAC,kBAAD,EAAqB,UAArB,CAH+B;AAIzCmB,IAAAA,SAAS,EAAEpB,MAAM,CACf,uBADe,EAEf,CAAC,MAAD,EAAS,YAAT,EAAuB,UAAvB,EAAmC,QAAnC,EAA6C,UAA7C,EAAyD,SAAzD,CAFe,EAGf,MAHe;AAJwB,GAA3C;AAUA,SACE,oBAAC,OAAD,EAAaS,YAAb,EACE,oBAAC,QAAD,EAAcQ,kBAAd,eADF,EAEE,oBAAC,QAAD,EAAcI,mBAAd,gBAFF,CADF;AAMD,CAtDH,EAuDE;AAAEC,EAAAA,IAAI,EAAEhB;AAAR,CAvDF","sourcesContent":["import { boolean, number, select, text } from '@storybook/addon-knobs'\nimport { storiesOf } from '@storybook/react'\nimport React from 'react'\nimport { FlexBox, FlexBoxProps, FlexItem, FlexItemProps } from './FlexBox'\nimport readme from './FlexBox.readme.md'\n\nstoriesOf('Design System/Helper/FlexBox', module).add(\n  'Default',\n  () => {\n    const flexBoxKnobs: FlexBoxProps = {\n      debug: boolean('box debug', true),\n      display: select('box display', ['flex', 'inline-flex'], 'flex'),\n      flexDirection: select(\n        'box flexDirection',\n        ['row', 'row-reverse', 'column', 'column-reverse'],\n        'row'\n      ),\n      flexWrap: select('box:flexWrap', ['nowrap', 'wrap', 'wrap-reverse'], 'wrap'),\n      justifyContent: select(\n        'box justifyContent',\n        ['flex-start', 'flex-end', 'center', 'space-between', 'space-around'],\n        'flex-start'\n      ),\n      alignContent: select(\n        'box alignContent',\n        ['flex-start', 'flex-end', 'center', 'baseline', 'stretch'],\n        'flex-start'\n      ),\n      alignItems: select(\n        'box alignItems',\n        ['flex-start', 'flex-end', 'center', 'space-between', 'space-around', 'stretch'],\n        'flex-start'\n      ),\n    }\n    const firstFlexItemProps: FlexItemProps = {\n      debug: boolean('first_item debug', true),\n      order: number('first_item order', 0),\n      flex: text('first_item flex', '0 1 auto'),\n      alignSelf: select(\n        'first_item alignSelf',\n        ['auto', 'flex-start', 'flex-end', 'center', 'baseline', 'stretch'],\n        'auto'\n      ),\n    }\n    const secondFlexItemProps: FlexItemProps = {\n      debug: boolean('second_item debug', true),\n      order: number('second_item order', 0),\n      flex: text('second_item flex', '0 1 auto'),\n      alignSelf: select(\n        'second_item alignSelf',\n        ['auto', 'flex-start', 'flex-end', 'center', 'baseline', 'stretch'],\n        'auto'\n      ),\n    }\n    return (\n      <FlexBox {...flexBoxKnobs}>\n        <FlexItem {...firstFlexItemProps}>First Item</FlexItem>\n        <FlexItem {...secondFlexItemProps}>Second Item</FlexItem>\n      </FlexBox>\n    )\n  },\n  { info: readme }\n)\n"],"file":"FlexBox.story.js"}