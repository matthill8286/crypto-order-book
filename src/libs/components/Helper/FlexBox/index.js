export * from "./FlexBox";
//# sourceMappingURL=index.js.map