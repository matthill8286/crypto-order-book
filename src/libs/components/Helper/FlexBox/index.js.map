{"version":3,"sources":["../../../../src/components/Helper/FlexBox/index.ts"],"names":[],"mappings":"AAAA","sourcesContent":["export * from './FlexBox'\n"],"file":"index.js"}