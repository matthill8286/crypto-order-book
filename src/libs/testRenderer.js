import { mount } from 'enzyme';
import React from 'react';
import { MemoryRouter as Router } from 'react-router-dom';
import renderer from 'react-test-renderer';
import { ThemeProvider } from 'styled-components';
import { saiyanTheme } from "./styles/sc-vars-saiyan";
export function renderWithTheme(tree) {
  var theme = arguments.length > 1 && arguments[1] !== undefined ? arguments[1] : saiyanTheme;
  return renderer.create(React.createElement(ThemeProvider, {
    theme: theme
  }, tree)).toJSON();
}
export function renderWithThemeAndRouter(tree) {
  var theme = arguments.length > 1 && arguments[1] !== undefined ? arguments[1] : saiyanTheme;
  return renderer.create(React.createElement(ThemeProvider, {
    theme: theme
  }, React.createElement(Router, null, tree))).toJSON();
}
export function renderWithRouter(tree) {
  return renderer.create(React.createElement(Router, null, tree)).toJSON();
}
export function mountWithTheme(tree) {
  return mount(React.createElement(ThemeProvider, {
    theme: saiyanTheme
  }, tree));
}
export function mountWithThemeAndRouter(tree) {
  return mount(React.createElement(ThemeProvider, {
    theme: saiyanTheme
  }, React.createElement(Router, null, tree)));
}
//# sourceMappingURL=testRenderer.js.map