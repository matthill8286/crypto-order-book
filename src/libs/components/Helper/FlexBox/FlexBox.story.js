import { boolean, number, select, text } from '@storybook/addon-knobs';
import { storiesOf } from '@storybook/react';
import React from 'react';
import { FlexBox, FlexItem } from "./FlexBox";
import readme from "./FlexBox.readme.md";
storiesOf('Design System/Helper/FlexBox', module).add('Default', function () {
  var flexBoxKnobs = {
    debug: boolean('box debug', true),
    display: select('box display', ['flex', 'inline-flex'], 'flex'),
    flexDirection: select('box flexDirection', ['row', 'row-reverse', 'column', 'column-reverse'], 'row'),
    flexWrap: select('box:flexWrap', ['nowrap', 'wrap', 'wrap-reverse'], 'wrap'),
    justifyContent: select('box justifyContent', ['flex-start', 'flex-end', 'center', 'space-between', 'space-around'], 'flex-start'),
    alignContent: select('box alignContent', ['flex-start', 'flex-end', 'center', 'baseline', 'stretch'], 'flex-start'),
    alignItems: select('box alignItems', ['flex-start', 'flex-end', 'center', 'space-between', 'space-around', 'stretch'], 'flex-start')
  };
  var firstFlexItemProps = {
    debug: boolean('first_item debug', true),
    order: number('first_item order', 0),
    flex: text('first_item flex', '0 1 auto'),
    alignSelf: select('first_item alignSelf', ['auto', 'flex-start', 'flex-end', 'center', 'baseline', 'stretch'], 'auto')
  };
  var secondFlexItemProps = {
    debug: boolean('second_item debug', true),
    order: number('second_item order', 0),
    flex: text('second_item flex', '0 1 auto'),
    alignSelf: select('second_item alignSelf', ['auto', 'flex-start', 'flex-end', 'center', 'baseline', 'stretch'], 'auto')
  };
  return React.createElement(FlexBox, flexBoxKnobs, React.createElement(FlexItem, firstFlexItemProps, "First Item"), React.createElement(FlexItem, secondFlexItemProps, "Second Item"));
}, {
  info: readme
});
//# sourceMappingURL=FlexBox.story.js.map