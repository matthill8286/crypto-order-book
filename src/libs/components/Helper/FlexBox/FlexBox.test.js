function _extends() { _extends = Object.assign || function (target) { for (var i = 1; i < arguments.length; i++) { var source = arguments[i]; for (var key in source) { if (Object.prototype.hasOwnProperty.call(source, key)) { target[key] = source[key]; } } } return target; }; return _extends.apply(this, arguments); }

import * as React from 'react';
import { renderWithTheme } from "../../../testRenderer";
import { FlexBox, FlexItem } from "./FlexBox";
describe('FlexBox', function () {
  var props = {};
  it('renders correctly', function () {
    var tree = renderWithTheme(React.createElement(FlexBox, props));
    expect(tree).toMatchSnapshot();
  });
  it('renders with direction column', function () {
    var tree = renderWithTheme(React.createElement(FlexBox, _extends({}, props, {
      flexDirection: "column"
    })));
    expect(tree).toMatchSnapshot();
  });
  it('renders with flex wrap', function () {
    var tree = renderWithTheme(React.createElement(FlexBox, _extends({}, props, {
      flexWrap: "wrap"
    })));
    expect(tree).toMatchSnapshot();
  });
});
describe('FlexItem', function () {
  var props = {};
  it('renders correctly', function () {
    var tree = renderWithTheme(React.createElement(FlexItem, props));
    expect(tree).toMatchSnapshot();
  });
});
//# sourceMappingURL=FlexBox.test.js.map