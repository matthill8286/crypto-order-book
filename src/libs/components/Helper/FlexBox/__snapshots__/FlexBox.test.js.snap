// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FlexBox renders correctly 1`] = `
<div
  className="FlexBox__StyledBox-sc-52meld-0 fkNwwV"
  display="flex"
/>
`;

exports[`FlexBox renders with direction column 1`] = `
<div
  className="FlexBox__StyledBox-sc-52meld-0 kSmAcZ"
  display="flex"
/>
`;

exports[`FlexBox renders with flex wrap 1`] = `
<div
  className="FlexBox__StyledBox-sc-52meld-0 hVXbHd"
  display="flex"
/>
`;

exports[`FlexItem renders correctly 1`] = `
<div
  className="FlexBox__StyledFlexItem-sc-52meld-1 cuJgpq"
  order={0}
/>
`;
